test/e2e/recipe/normalize.ts

```diff
@@ -75,6 +80,38 @@ function normalizeEvent(event: Event): Event {
     event.contexts.device.arch = '{{arch}}';
   }
 
+  if (event.contexts?.device?.memory_size) {
+    event.contexts.device.memory_size = 0;
+  }
+
+  if (event.contexts?.device?.free_memory) {
+    event.contexts.device.free_memory = 0;
+  }
+
+  if (event.contexts?.device?.processor_count) {
+    event.contexts.device.processor_count = 0;
```

Comment:
Would be nice to have unit tests to sanity check this integration. Perhaps we can make a big TODO issue and open it up as a `good first issue` for the community (or a future Sentry intern 😄)

Reply:
We could certainly have better normalisation of events. For example, check that everything we normalise is at least reasonable and close to what's expected.

`chai-subset` makes the event comparisons 1 line but we could certainly have something better!

Reply:
Can we have assertions similar to `expect.any(String)` or `expect.any(Number)` before we apply the normalization to check? Not convinced its the best idea though.

Reply:
We need an object comparison/diffing library that lets us put ranges or custom validation functions...

```json
  "timestamp": "validate:timestamp",
```

Reply:
It looks like `chai-subset` already supports functions in the expected event:
https://github.com/debitoor/chai-subset/blob/5c943ada7356c23d3bd9b04ba2b230ec8a5eebac/lib/chai-subset.js#L74-L76

Just need to create a JSON revive function that replaces validation strings with the correct functions!